### PR TITLE
Set the error status code for every SqliteException instance

### PR DESCRIPTION
**Pull request by AndrejMitrovic**
Also fixed the tests which were broken.

We still need to fix the `dub test` target, and also set up a `CI`. But I'll do that in another PR.

### CODE REVIEW

**Comment by Geod24** [comment]
Rebase plz

**Comment by AndrejMitrovic** [comment]
yep

File: source/d2sqlite3/database.d [hunk 1]
```diff
@@ -941,7 +941,7 @@ public:
     /++
     Registers a delegate of type `TraceCallbackDelegate` as the trace callback.
 
-    Any previously set trace callback is released.
+    Any previously set profile or trace callback is released.
```
Comment (Geod24) [inline source/d2sqlite3/database.d:952]: First commit message: "same tim".

**Comment by AndrejMitrovic** [comment]
Rebased.

File: source/d2sqlite3/database.d [hunk 2]
```diff
@@ -377,9 +377,17 @@ public:
     void loadExtension(string path, string entryPoint = null)
     {
         assert(p.handle);
-        immutable ret = sqlite3_load_extension(p.handle, path.toStringz, entryPoint.toStringz, null);
+        char* p_err;
+        scope (failure)
+        {
+            if (p_err !is null)
+                sqlite3_free(p_err);
```
Comment (Geod24) [inline source/d2sqlite3/database.d]: The check is not needed, `sqlite3_free` behaves like `free` (`NULL` => no-op)

**Review by Geod24 — APPROVED** [review]
Just a few nits, feel free to merge once fixed

**Pull request merged**